### PR TITLE
CI: add HIP tests 

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- add hipcc image definition
- activate tests for hip
- `n_wise_generator.py`:
  - fix issue that `clang++` was not detected
- remove unused file `share/ci/generate.sh`
-  `share/ci/compiler_clang.yml` removed unused alpaka cmake flag `DALPAKA_CUDA_COMPILER`

Currently, we do only a compile test for the PMacc unit tests. For unknown reasons, all runtime tests on our AMD ROCm CI fail after the last restart of the CI node. On our AMD dev system, all tests passing. This behavior requires further investigations but should not block us from testing PIConGPU with HIP.

### CODE REVIEW

**Review by PrometheusPi — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
ready for review

File: share/ci/n_wise_generator.py [hunk 1]
```diff
@@ -56,14 +61,30 @@ def is_valid_combination(row):
...
         is_nvcc = True if len(row[0]) == 3 and row[0][2] == "nvcc" else False
         is_cuda = True if row[1][0] == "cuda" else False
         v_cuda = get_version(row[1])
 
+        # hipcc
+        is_hipcc = True if len(row[0]) == 3 and row[0][2] == "hipcc" else False
+        is_hip = True if row[1][0] == "hip" else False
+
+        # CI nvcc image is not shipped with clang++
+        # clang_cuda images can currenlty not be used because
```
Comment (sbastrakov) [inline share/ci/n_wise_generator.py:77]: Typo: should be `currently`

File: share/ci/n_wise_generator.py [hunk 2]
```diff
@@ -56,14 +61,30 @@ def is_valid_combination(row):
...
+        # the base image is hard coding -DALPAKA_CUDA_COMPILER=clang
```
Comment (sbastrakov) [inline share/ci/n_wise_generator.py:78]: Do you mean `hard setting` ?

**Review by sbastrakov — CHANGES_REQUESTED** [review]
Looks good to me, just 2 small text suggestions.

File: share/ci/n_wise_generator.py (hunk 2 above)
Comment (psychocoderHPC) [reply share/ci/n_wise_generator.py:78]: will change it to `... is setting ...`

**Comment by psychocoderHPC** [comment]
Simeon provided the ROCm 4.0 container, therefore I will close this PR in favor of #3538. The new ROCm container also solved the runtime issue I observed on the AMD CI nodes.

@sbastrakov Your suggestions are included in #3538

**Pull request closed without merging**